app/validation/validators.py

```diff
@@ -345,3 +345,37 @@ def format_date_string(field):
     return '{}-{:02d}-{:02d}'.format(int(field.data['year']),
                                      int(field.data['month']),
                                      day)
+
+
+class MutuallyExclusive(object):
+    def __init__(self, options, messages=None):
+        self.exclusive_value = options[-1]['value']
+        self.value_labels = self._get_value_labels(options)
+        self.messages = messages or error_messages
+
+    def __call__(self, form, field):
+        values = field.data
+        if (len(values) > 1) and (self.exclusive_value in values):
+            non_exclusives = self._build_non_exclusives(values.copy())
+            raise validators.ValidationError(self.messages['MUTUALLY_EXCLUSIVE'] %
+                                             dict(non_exclusives=non_exclusives, exclusive=self.value_labels[self.exclusive_value]))
+
+    def _build_non_exclusives(self, values):
+        values.remove(self.exclusive_value)
+        non_exclusives = ''
+        for index, additional in enumerate(values):
```

Comment:
Might be able to reduce this logic to:

```
non_exclusive_values = [
    '"{}"'.format(self.value_labels[value])
    for value in values
    if value != self.exclusive_value
]
non_exclusives = "{} and {}".format(", ".join(non_exclusive_values[:-1]),  non_exclusive_values[-1])
```

Reply:
The `non_exclusives` wouldn't work where only one non-exclusive value is selected but I do prefer the building of non_exclusive_values.

Reply:
Good compromise 👍